src/ray/raylet/main.cc

```
@@ -291,6 +299,20 @@ int main(int argc, char *argv[]) {
         << "Resource isolation with cgroups is only supported in linux. Please set "
            "enable_resource_isolation to false. This is likely a misconfiguration.";
 #endif
+
+    // Move system processes into the system cgroup.
+    std::vector<std::string> system_pids_to_move = absl::StrSplit(system_pids, ",");
+    system_pids_to_move.emplace_back(std::to_string(ray::GetPID()));
+    for (const auto &pid : system_pids_to_move) {
+      ray::Status s = cgroup_manager->AddProcessToSystemCgroup(pid);
+      // TODO(#54703): This could be upgraded to a RAY_CHECK.
```

I'm not sure if this should be a RAY_CHECK or not yet. CgroupManager::AddProcessToSystemCgroup already fails a RAY_CHECK if the cgroup doesn't exist or permissions are wrong. The only other errors that are possible here are if the pid is invalid or the process doesn't exist.

I think there's a very strong case to be made that if a system process does not exist when you try to move it into the cgroup, we should fail fast.

---

Considering that pids are passed to the raylet by argument I think the case for ray_check is strong.  Let's do it.

---

+1, `RAY_CHECK` it boss (with a good error)